lib/self-evict.js

```diff
@@ -177,10 +179,57 @@ SelfEvict.prototype._preEvict = function _preEvict() {
 };
 
 SelfEvict.prototype._evict = function _evict() {
-    this._transitionTo(PhaseNames.Evicting);
+    var self = this;
+    var ringpop = self.ringpop;
+    var phase = this._transitionTo(PhaseNames.Evicting);
+
+    ringpop.membership.setLocalStatus(Member.Status.faulty);
+
+    var pingEnabled = this.ringpop.config.get('selfEvictionPingEnabled');
+    if (pingEnabled) {
+        var selfEvictionMaxPingRatio = ringpop.config.get('selfEvictionMaxPingRatio');
+        var membership = ringpop.membership;
+
+        var pingableMembers = _.filter(membership.members, membership.isPingable.bind(membership));
+        var numberOfPingableMembers = pingableMembers.length;
+        var maxNumberOfPings = Math.ceil(numberOfPingableMembers * selfEvictionMaxPingRatio);
+
+        // Never ping more than the maximum piggyback counter and/or number of pingable members.
+        var numberOfPings = Math.min(
```

Comment:
I found it hard to read through this block of code. A lot is happening, for both peer selection and calculation of how many we should ping. For readability and testability I would love to find a way to pull this logics into its own function.

I think we can do so by returning the list of all candidates that will be pinged and use `list.length` when we need to get the number of nodes we are planning to ping.